### ISSUE TITLE
TFile::Open(), HTTP basic authentication and DaviX

### ISSUE THREAD

**Issue by cozzyd** [root-project/root#9072]
### Describe the bug
If DAVIX is installed, it is used for TFile::Open("https://...") likes, but unlike TWebFile, it does not seem to support basic auth. 

### Expected behavior
auto f = TFile::Open("https://user:password@example.org/lotterynumbers.root") should work regardless of if DaviX is installed or not. 

### To Reproduce

Try to access a file hidden behind http basic authentication using TFile::Open with and without DaviX. Without DaviX installed, TFile::Open( ) delegates to TWebFile and it works. With DaviX, DaviX tries to handle it but it doesn't support basic auth. 

### Setup
root 6.24/04 on Fedora 34 (x86-64) installed via dnf, with and without root-net-davix. 

### Additional context
Note that there is some functionality supported by DaviX not supported by TWebFile (see e.g. #8890), so delegating to TWebFile is not necessarily the right solution. Probably better to add basic auth support to Davix (with the same TUrl parsing of passwords, so that e.g. escaping @ remains the same).

**Comment by Axel-Naumann** [comment]
Yes looks good, @mpatrascoiu ! I see you have tests - I don't think we can add much here :-)

**Comment by mpatrascoiu** [comment]
Hello,

Given the uncertain future of Davix, I don't think this issue will get addressed.
Also, with the curl backend for Davix, basic authentication is not supported altogether (see https://github.com/cern-fts/davix/issues/81)

It's better to think of workarounds in the ROOT level.

**Comment by dpiparo** [comment]
Unfortunately, without proper Davix support, there won't be workarounds at the ROOT level.
